perf(allocator): `Allocator::from_raw_parts` get offset of chunk footer field as const

overlookmotel opened the pull request:
`Allocator::from_raw_parts` manually amends `bumpalo::Bump` to inject a manually-created allocator chunk. To do this it needs to locate the `current_chunk_footer` field of `Bump`. Remove some unnecessary assertions, which allows compiler to const fold the whole process down to a static integer.

camc314 commented on crates/oxc_allocator/src/from_raw_parts.rs:
```diff
@@ -200,6 +150,58 @@ struct ChunkFooter {
+/// This should all be const-folded down by compiler.
+/// https://godbolt.org/z/eKdMcdEYa
+/// `#[inline(always)]` because this is essentially a const function
+#[expect(clippy::inline_always)]
```
ahaha!! It comments this every time we use `expect`!!

overlookmotel commented:
I don't think anyone else is really across this jiggery-pokery, so merging without review.

overlookmotel committed:
perf(allocator): `Allocator::from_raw_parts` get offset of chunk footer field as const (#12080)

`Allocator::from_raw_parts` manually amends `bumpalo::Bump` to inject a manually-created allocator chunk. To do this it needs to locate the `current_chunk_footer` field of `Bump`. Remove some unnecessary assertions, which allows compiler to const fold the whole process down to a static integer.
